### DISCUSSION TITLE
Permission required to use query_vulnerability

### DISCUSSION THREAD

**Discussion by aosti** [CrowdStrike/falconpy#1238] [Q&A] [answered]
Hello FalconPy Community,

I am encountering an issue with the Falcon Intel API when using FalconPy and was hoping for some guidance or suggestions on how to resolve it.

I have a falcon client that works for getting the actors, but when I attempt to use the query_vulnerability method, it returns an error 403. Which permission do I need to execute this operation?

**Comment by jshcodes** [comment]
Hi @aosti -

Thank you for the question!  

Does your key have the **Vulnerabilities (Falcon Intelligence)** scope assigned?

**Comment by poncion** [comment]
Hello, I’m collaborating with Aosti in this case, and we are using the Vulnerabilities (Falcon Intelligence) scope.
'code': 403, 'message': 'access denied, authorization failed'

However, we do not experience this error when using other API methods, such as `query_actor_entities`.

Our code reference:
https://www.falconpy.io/Service-Collections/Intel.html#queryvulnerabilities

**Reply by jshcodes** [reply]
I get a 200 with results from the [QueryVulnerabilities](https://www.falconpy.io/Service-Collections/Intel.html#queryvulnerabilities) operation using a key with only the one scope assigned in my test environment.

Can we confirm we have this scope assigned:
![Screenshot 2024-10-29 at 2 08 13 AM](https://github.com/user-attachments/assets/b6b65593-71fa-48e1-8e57-58ab577af5eb)

and not this one?
![Screenshot 2024-10-29 at 2 07 59 AM](https://github.com/user-attachments/assets/50642c7b-cf27-477b-b7b7-946bd8b7dd9c)


Are you using the [Service Class](https://www.falconpy.io/Usage/Basic-Service-Class-usage.html) [example](https://www.falconpy.io/Service-Collections/Intel.html#service-class-example-pep8-syntax-21) or the [Uber Class](https://www.falconpy.io/Usage/Basic-Uber-Class-usage.html) [example](https://www.falconpy.io/Service-Collections/Intel.html#uber-class-example-21)?

**Reply by poncion** [reply]
I can't see the "Vulnerabilities (Falcon Intelligence)" scope in my /api-clients-and-keys. I can see "vulnerabilities" and other Falcon Intelligence scopes. Here’s my list of my scopes in use:

**Read Only:**
- Vulnerabilities  
- Actors (Falcon Intelligence)  
- Indicators (Falcon Intelligence)  
- Malware Families (Falcon Intelligence)  
- Reports (Falcon Intelligence)  
- Sandbox (Falcon Intelligence)  
- Quick Scan (Falcon Intelligence)  

I'm trying to use query_vulnerabilities because I'd like to list all vulnerabilities with specific adversaries.


I'm using Service Classe:
```
from falconpy.intel import Intel
falcon = Intel(client_id=CLIENT_ID, client_secret=CLIENT_SECRET)

response = falcon.query_vulnerabilities()
print(response)
```

**Reply by jshcodes** [reply, marked as answer]
The scope should have `(Falcon Intelligence)` in the name like the others.  If `Vulnerabilities (Falcon Intelligence)` is not one of the available scope options, you may need to reach out to the [Support Team](https://supportportal.crowdstrike.com/) to check which products are available in your tenant.
